### ISSUE TITLE
SwashbuckleAspNetVersioningShim.RemoveVersionParametersOperationFilter.Apply(Operation operation, OperationFilterContext context) throws System.InvalidOperationException: Sequence contains more than one matching element

### ISSUE THREAD

**Issue by dlp211** [rh072005/SwashbuckleAspNetVersioningShim#6]
Have three controlers, all with `[ApiVersion("1")]` attribute set and all controller routes have the form `[Route("api/v{version:apiVersion}/thing/[controller]")]`

Startup.cs:

```
public class Startup
    {
        public Startup(IHostingEnvironment env)
        {
            var builder = new ConfigurationBuilder()
                .SetBasePath(env.ContentRootPath)
                .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
                .AddEnvironmentVariables();
            Configuration = builder.Build();
        }

        public IConfigurationRoot Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            // Add framework services.
            var mvcBuilder = services.AddMvc();
            services.AddApiVersioning();
            services.AddSwaggerGen(opt =>
            {
                SwaggerVersioner.ConfigureSwaggerGen(opt, mvcBuilder.PartManager);
                //opt.IncludeXmlComments(GetXmlCommentsPath());
            });
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory, ApplicationPartManager partManager)
        {
            loggerFactory.AddConsole(Configuration.GetSection("Logging"));
            loggerFactory.AddDebug();

            app.UseMvc();
            app.UseSwagger();
            app.UseSwaggerUI(c =>
            {
                SwaggerVersioner.ConfigureSwaggerUI(c, partManager);
            });
        }

        private static string GetXmlCommentsPath()
        {
            var app = PlatformServices.Default.Application;
            return Path.Combine(app.ApplicationBasePath, Path.ChangeExtension(app.ApplicationName, "xml"));
        }
    }
```

Stack Trace:

```
System.InvalidOperationException: Sequence contains more than one matching element
   at System.Linq.Enumerable.Single[TSource](IEnumerable`1 source, Func`2 predicate)
   at SwashbuckleAspNetVersioningShim.RemoveVersionParametersOperationFilter.Apply(Operation operation, OperationFilterContext context)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreateOperation(ApiDescription apiDescription, ISchemaRegistry schemaRegistry)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.CreatePathItem(IEnumerable`1 apiDescriptions, ISchemaRegistry schemaRegistry)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.<>c__DisplayClass4_0.<GetSwagger>b__5(IGrouping`2 group)
   at System.Linq.Enumerable.ToDictionary[TSource,TKey,TElement](IEnumerable`1 source, Func`2 keySelector, Func`2 elementSelector, IEqualityComparer`1 comparer)
   at Swashbuckle.AspNetCore.SwaggerGen.SwaggerGenerator.GetSwagger(String documentName, String host, String basePath, String[] schemes)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.<Invoke>d__6.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Builder.RouterMiddleware.<Invoke>d__4.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Server.IISIntegration.IISMiddleware.<Invoke>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.AspNetCore.Hosting.Internal.RequestServicesContainerMiddleware.<Invoke>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.Frame`1.<RequestProcessingAsync>d__2.MoveNext()
```

**Comment by dlp211** [comment]
I figured out the source of the issue, but I'm not sure of the correct way to fix this.  The issue is that one of the endpoints in my controller has a query parameter name `version`.  This causes `.Single` to fail with the above exception.

**Comment by rh072005** [comment]
Hi @dlp211 
Do you have an example of the controller code you can share that produces this issue? 
I tried it out with a ``version`` parameter on a get method, it removes that parameter but doesn't seem to throw the exception you saw. 
This is what I've tried:
```csharp
[ApiVersion("1")]
[Route("api/v{version:apiVersion}/thing/[controller]")]
public class ValuesController : Controller
{
    // GET api/values
    [HttpGet]
    public IEnumerable<string> Get(string version)
    {
        return new string[] { "value1", "value2" };
    }
```

**Comment by dlp211** [comment]
```
 [ApiVersion("1")]
 [Route("api/v{version:apiVersion}/thing/[controller]")]
 public class Authoring : Controller
 {
        [Route("{measureId}")]
        [HttpGet]
        public IActionResult Get([FromRoute] int measureId, [FromQuery] int? version = null, [FromQuery] int? iteration = null)
        {
            return Ok();
         }
}
```

**Comment by rh072005** [comment]
Cheers, I've managed to recreate that issue now.
The operation filter at the minute looks for version parameters regardless of where they are found. 
The version parameter from the API Versioning library is in the path (in this implementation) but there was no check for that specifically and it found that one and the one in the query. 
I've just pushed 0.4.3-pre that fixes this (i.e. it only looks in the path), give it a try and let me know if it works for you.
